### PR TITLE
Egonet a.k.a. per-edge neighbors

### LINKED ISSUE #32: neighbors
I was thinking that, when doing simulations (particularly for games), it is very useful to have the neighbors of a node divided by hyperedge. To explain better, here's what the current `neighbors` function returns:
```python
hyperedge_list = [[1, 2], [2, 3, 4]]
H.neighbors(2)
# {1, 3, 4}
```
which is basically the neighbors in the graph projection of H. What I was imagining is actually a higher-order neighbor function that returns ` {[1], [3, 4]}`. This is extremely useful in loops when you want to scroll over the hyperedges attached to a node and then the neighbors in those.

Do you think something like this could be integrated into the current `neighbors` function? Or maybe two different ones? In my old code I had `H.graph_neighbors()` and `H.neighbors()`.

### PR DESCRIPTION
**Pull request by leotrs**
I'm moving the conversation from #32 to here.

The current is a proof of concept. We just have to agree on the name and whether we want an analogous method for edges. Once we agree on that, I'll implement those and add tests.

Demo:
```python
In [6]: H = xgi.Hypergraph([(0,1,2), (1,2), (0, 3)])

In [7]: H.neighbors(0)
Out[7]: {1, 2, 3}

In [8]: H.egonet(0)
Out[8]: [[1, 2], [3]]

In [9]: H.egonet(0, include_self=True)
Out[9]: [[0, 1, 2], [0, 3]]
```

Closes #32.

### CODE REVIEW

**Review by iaciac — APPROVED** [review]
Looks clean to me! :)

**Comment by maximelucas** [comment]
Looks good. Do we want `H.neighbors(0)` to return a set or a list (of unique values)?

**Comment by leotrs** [comment]
I think a set is more meaningful because it makes it more readily apparent that each neighbor appears only once.

**Comment by leotrs** [comment]
Alright since you both think it's good - this is ready for review. Any thoughts on the name of the method?

File: xgi/classes/hypergraph.py [hunk 1]
```diff
@@ -209,6 +209,43 @@ def neighbors(self, n):
...
+        n : node
+            Node whose egonet is needed.
+        include_self : bool (default False)
+            Whether the egonet contains `n`.
+
+        Returns
+        -------
+        list
+            An edgelist of the egonet of `n`.
+
```
[the remark this answers is not in the document]
Comment (leotrs) [reply xgi/classes/hypergraph.py:235]: Done!

**Comment by maximelucas** [comment]
I think it's good. Regarding the name, I would not intuitively go looking for `H.egonet()` when wanting its output. But I think since we linked both functions in the doc, it's okay. 

Do we want to add the `order` parameter in both of those? Can be later too. It already works as it is.

**Comment by leotrs** [comment]
Oh good call - that makes me think we might want to leave this for later so we may have a systematic way of filtering edges by order across the entire codebase...

**Review by maximelucas — APPROVED** [review]

**Pull request merged**